lib/galaxy/model/store/ro_crate_utils.py

```diff
@@ -67,88 +85,148 @@ def __init__(self, model_store: Any):
             "dbkey",
             "__input_ext",
         ]
+        self.workflow_entities: Dict[int, Any] = {}
+        self.collection_entities: Dict[int, Any] = {}
+        self.file_entities: Dict[int, Any] = {}
+        self.param_entities: Dict[int, Any] = {}
         self.pv_entities: Dict[str, Any] = {}
 
     def build_crate(self):
         crate = ROCrate()
-        file_entities = self._add_files(crate)
-        self._add_collections(crate, file_entities)
         self._add_workflows(crate)
         self._add_engine_run(crate)
-        self._add_actions(crate, file_entities)
+        self._add_create_action(crate)
+        self._add_collections(crate)
+        self._add_files(crate)
+        self._add_profiles(crate)
+        self._add_parameters(crate)
+        self._add_attrs_files(crate)
         return crate
 
-    def _add_files(self, crate: ROCrate) -> Dict[int, Any]:
-        file_entities: Dict[int, Any] = {}
-        for dataset, _ in self.model_store.included_datasets.values():
-            if dataset.dataset.id in self.model_store.dataset_id_to_path:
-                filename, _ = self.model_store.dataset_id_to_path[dataset.dataset.id]
-                if not filename:
-                    filename = f"datasets/dataset_{dataset.dataset.uuid}"
-                name = dataset.name
-                encoding_format = dataset.datatype.get_mime()
+    def _add_file(self, dataset: HistoryDatasetAssociation, properties: Dict[Any, Any], crate: ROCrate) -> File:
+        if dataset.dataset.id in self.model_store.dataset_id_to_path:
+            filename, _ = self.model_store.dataset_id_to_path[dataset.dataset.id]
+            if not filename:
+                filename = f"datasets/dataset_{str(dataset.dataset.uuid)}"
```

Comment:
I would expect the str casting to be redundant here, don't f-strings take care of that directly?

Reply:
Yep, that works fine.
```suggestion
                filename = f"datasets/dataset_{dataset.dataset.uuid}"
```